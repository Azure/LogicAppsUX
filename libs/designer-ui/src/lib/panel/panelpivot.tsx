import constants from '../constants';
import type { PageActionTelemetryData } from '../telemetry/models';
import type { PanelTab } from './panelUtil';
import type { IPivotStyles } from '@fluentui/react/lib/Pivot';
import { Pivot, PivotItem } from '@fluentui/react/lib/Pivot';
import React from 'react';
import { useIntl } from 'react-intl';

const pivotStyles: Partial<IPivotStyles> = {
  link: {
    margin: '1px 24px 1px 1px',
  },
  text: {
    '&:hover': {
      color: constants.PANEL_HIGHLIGHT_COLOR,
    },
  },
};
export interface CategoryPivotProps {
  isCollapsed: boolean;
  tabs: Record<string, PanelTab>;
  selectedTab?: string;
  onTabChange(tabName?: string): void;
  onCategoryClick?(item: PivotItem): void;
  trackEvent(data: PageActionTelemetryData): void;
}
export const PanelPivot = ({ isCollapsed, tabs, selectedTab, onTabChange, trackEvent }: CategoryPivotProps): JSX.Element => {
  const intl = useIntl();
  const onTabSelected = (item?: PivotItem): void => {
    if (item) {
      const { itemKey } = item.props;
      trackEventHandler(isCollapsed, itemKey);
      onTabChange(itemKey);
    }
  };

  const trackEventHandler = (isCollapsed: boolean, itemKey?: string): void => {
    // TODO: 12798935 Analytics (event logging)
  };
  const overflowLabel = intl.formatMessage({
    defaultMessage: 'more panels',
    description: 'This is a label to access the overflowed panels',
  });
  return (
    <div className="msla-pivot">
      <Pivot
        styles={pivotStyles}
        selectedKey={selectedTab}
        className="msla-panel-menu"
        onLinkClick={onTabSelected}
        overflowBehavior="menu"
        overflowAriaLabel={overflowLabel}
      >
        {Object.entries(tabs).map(([name, PanelTab]) => {
          return <PivotItem key={name} itemKey={name} headerText={PanelTab.title} />;
        })}
      </Pivot>
    </div>
  );
};
