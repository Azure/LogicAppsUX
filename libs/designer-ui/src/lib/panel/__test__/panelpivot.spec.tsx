import type { PanelPivotProps } from '../panelpivot';
import { PanelPivot } from '../panelpivot';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';

describe('ui/workflowparameters/workflowparameter', () => {
  let minimal: PanelPivotProps, renderer: ReactShallowRenderer.ShallowRenderer;
  beforeEach(() => {
    minimal = {
      nodeId: '',
      isCollapsed: false,
      tabs: {},
      onTabChange: jest.fn(),
      trackEvent: jest.fn(),
    };
    renderer = ReactShallowRenderer.createRenderer();
  });

  afterEach(() => {
    renderer.unmount();
  });

  it('should construct.', () => {
    const panelPivot = renderer.render(<PanelPivot {...minimal} />);
    expect(panelPivot).toMatchSnapshot();
  });

  it('should render when no tabs.', () => {
    renderer.render(<PanelPivot {...minimal} />);
    const panelPivot = renderer.getRenderOutput();

    expect(panelPivot.props.className).toBe('msla-pivot');

    const pivotMenu = panelPivot.props.children;

    expect(pivotMenu.props.className).toBe('msla-panel-menu');
    expect(pivotMenu.props.overflowAriaLabel).toBe('more panels');
    expect(pivotMenu.props.children).toHaveLength(Object.keys(minimal.tabs).length);
  });

  it('should render menu items when passed tabs.', () => {
    // TODO: 13865562 When Tabs get setup
  });
});
