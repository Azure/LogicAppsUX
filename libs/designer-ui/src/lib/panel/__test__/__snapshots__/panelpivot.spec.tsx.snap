// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/workflowparameters/workflowparameter should construct. 1`] = `
<div
  id="msla-node-details-panel-"
>
  <ForwardRef
    aria-label="more panels"
  >
    <TabList
      onTabSelect={[Function]}
      style={
        Object {
          "margin": "0px -12px",
        }
      }
    >
      <OverflowMenu
        tabs={Array []}
      />
    </TabList>
  </ForwardRef>
  <div
    className="msla-panel-content-container"
  />
</div>
`;
